src/components/TodoForm.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { TodosWithUsers } from '../type.ts/TodosWithUser';
+import { getUserById } from '../utils/userId';
+import { User } from '../type.ts/User';
+
+type Props = {
+  onAdd: (todoWithUser: TodosWithUsers) => void;
+  users: User[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onAdd, users }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const onUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleAdd = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onAdd({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId) || null,
```

Comment:
The `getUserById` function might return `null`. Ensure that this is handled properly to avoid rendering issues in the `TodoList` component.